<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <param name="use_sim_time" value="true"/>
    <arg name="bag_name" default="some.bag"/>
    <arg name="start_time" default="0"/>
    <node name="match_bag" pkg="rosbag" type="play" output="screen" args="$(find tj2_match_watcher)/bags/$(arg bag_name) --clock --pause -s $(arg start_time)"/>
    <include file="$(find tj2_description)/launch/tj2_description_2022_robot.launch"/>
    <include file="$(find tj2_target)/launch/tj2_target.launch"/>
    
    <node pkg="tj2_laser_slam" type="odom_bag_publisher.py" name="odom_bag_publisher" output="screen">
        <remap from="odom" to="/tj2/odom"/>
    </node>
    <include file="$(find tj2_rplidar)/launch/laser_combiner.launch"/>
    <group ns="sinistra_laser">
        <node pkg="laser_filters" type="scan_to_scan_filter_chain" name="sinistra_laser_filter">
            <rosparam command="load" file="$(find tj2_rplidar)/config/sinistra_laser_filter.yaml" />
        </node>
    </group>
    <group ns="dextra_laser">
        <node pkg="laser_filters" type="scan_to_scan_filter_chain" name="dextra_laser_filter">
            <rosparam command="load" file="$(find tj2_rplidar)/config/dextra_laser_filter.yaml" />
        </node>
    </group>

    <arg name="map_name" default="rapid-react-2022"/>
    <include file="$(find tj2_laser_slam)/launch/tj2_laser_slam.launch">
        <arg name="mode" value="localize"/>
        <arg name="map_name" value="$(arg map_name)"/>
    </include>
    <include file="$(find tj2_waypoints)/launch/tj2_waypoints.launch">
        <arg name="waypoints_path" value="$(find tj2_waypoints)/waypoints/$(arg map_name)"/>
    </include>
</launch>
